controllers/eventhandlers_test.go

```diff
@@ -475,3 +500,351 @@ func (q *DelegatingQueue) Forget(item interface{}) {}
 func (q *DelegatingQueue) NumRequeues(item interface{}) int {
 	return 0
 }
+
+func Test_enqueueOnDeletionRequestHandler_Delete(t *testing.T) {
+	t.Parallel()
+
+	ctx := context.Background()
+	kind := VaultStaticSecret
+	ownerRefsSupported := []metav1.OwnerReference{
+		{
+			APIVersion: secretsv1beta1.GroupVersion.String(),
+			Kind:       kind.String(),
+			Name:       "baz",
+		},
+	}
+
+	ownerRefsUnsupported := []metav1.OwnerReference{
+		{
+			APIVersion: secretsv1beta1.GroupVersion.String(),
+			Kind:       "Unknown",
+			Name:       "foo",
+		},
+	}
+	deleteEventSupported := event.DeleteEvent{
+		Object: &secretsv1beta1.SecretTransformation{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:       "default",
+				Name:            "vso-secret",
+				OwnerReferences: ownerRefsSupported,
+			},
+		},
+	}
+
+	deleteEventUnsupported := event.DeleteEvent{
+		Object: &secretsv1beta1.SecretTransformation{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace:       "default",
+				Name:            "vso-secret",
+				OwnerReferences: ownerRefsUnsupported,
+			},
+		},
+	}
+
+	wantAddedAfterValid := []any{
+		reconcile.Request{
+			NamespacedName: client.ObjectKey{
+				Namespace: "default",
+				Name:      "baz",
+			},
+		},
+	}
+
+	gvk := secretsv1beta1.GroupVersion.WithKind(kind.String())
+	tests := []testCaseEnqueueOnDeletionRequestHandler{
+		{
+			name: "enqueued",
+			kind: kind,
+			deleteEvents: []event.DeleteEvent{
+				deleteEventSupported,
+			},
+			q: &DelegatingQueue{
+				Interface: workqueue.New(),
+			},
+			gvk:             gvk,
+			wantAddedAfter:  wantAddedAfterValid,
+			maxRequeueAfter: time.Second * 10,
+		},
```

Comment:
It looks like `kind`, `q`, and `gvk` are invariant between the test cases, so they could be defined inside the t.Run to make these test case definitions smaller.

Reply:
I am kind of partial to having them as test input member for potential, future types and work queues.